smarts/core/sensors.py

```diff
@@ -244,10 +250,13 @@ def observe(sim, agent_id, sensor_state, vehicle) -> Tuple[Observation, bool]:
                 nv_lane = sim.road_map.nearest_lane(
                     nv.pose.point, radius=vehicle.length
                 )
+                nv_lane_pos = None
                 if nv_lane:
                     nv_road_id = nv_lane.road.road_id
                     nv_lane_id = nv_lane.lane_id
                     nv_lane_index = nv_lane.index
+                    if vehicle.subscribed_to_lane_position_sensor:
+                        nv_lane_pos = vehicle.lane_position_sensor(nv_lane, nv)
```

Comment:
This is a future issue but it feels like collecting neighborhood vehicle observations could be significantly optimized through a locally cached method.

Reply:
https://github.com/huawei-noah/SMARTS/issues/1414

Reply:
Yes, agreed, we've already seen that `offset_along_lane()` and `to_lane_coord()` are on the hot path in a few other contexts as well.  These will definitely need to be optimized as much as possible.  (This -- i.e., performance considerations -- is also the reason I made this a sensor that you can disable.)